### ISSUE TITLE
Executable file permission is lost when app is created from template

### ISSUE THREAD

**Issue by amithgeorge** [seancorfield/deps-new#22]
I am creating a new template for my projects. The root folder has a script at path `root/bin/kaocha` with executable permissions. 

When creating a dummy application, the script is created correctly. However it doesn't have the executable permission on it. All this is on WSL2 Linux.

Slack conversation - https://clojurians.slack.com/archives/C019ZQSPYG6/p1638439206035200

Link to repo for my template - https://github.com/amithgeorge/deps-new-app-template 

Command to create app from template - 

```shell
clojure -Sdeps '{:deps {com.amithgeorge/app-template {:git/sha "0db7f25ae1f17f5ee739704389853531ebc7ad6b", :git/url "https://github.com/amithgeorge/deps-new-app-template.git"}}}' -Tnew create :template com.amithgeorge/app-template :name com.example/test-app
```

**Other debugging steps**

`clojure.tools.build.api/copy-dir` is able to properly copy files and retain the executable permission. Repo - https://github.com/amithgeorge/tools-build-executable-file-copy-bug

**Comment by seancorfield** [comment]
Figured it out!

This is a weird one. If you run `b/copy-dir` with `:replace`, it reads source files, transforms them, and writes target files -- it doesn't "copy" them, so it doesn't preserve permissions. If you run `b/copy-dir` _without_ `:replace`, it just copies the files and so it preserves permissions.

I "fixed" it in your template by adding the following `:transform` line:

```clojure
["root" "." {"bin/kaocha" "bin/kaocha"} :raw :only]
```

It's a bit hacky because it recopies `bin/kaocha` without `:replace` over the existing one from `root` and it relies on not doing any text replacements in the file.

So I think this is really a bug in how `tools.build` handles text replacement in executable shell scripts and needs to be raised against that. I'll create an "ask" about it and add that back here for reference.

**Comment by seancorfield** [comment]
https://ask.clojure.org/index.php/11349/tools-build-copy-dir-with-replace-loses-file-permissions

If that gets fixed in `tools.build`, it will automatically work in `deps-new` so I'm closing this out (as "not a `deps-new` bug") -- and you have a partial workaround for now (as long as you don't need replacement text in your shell script).

Depending on what @puredanger thinks about that "ask" issue, I may decide to take some action in `deps-new` at a future date.

**Comment by amithgeorge** [comment]
I implemented your suggestion. It works. Thank you.

**Comment by seancorfield** [comment]
`tools.build` has been updated to fix TBUILD-24 and `deps-new` v0.4.5 includes that version (just released).
